src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -42,32 +52,175 @@ public class PipelineCacheServer
                     return;
                 }
 
-                //Upload the pipeline artifact.
-                PipelineCacheActionRecord uploadRecord = clientTelemetry.CreateRecord<PipelineCacheActionRecord>((level, uri, type) =>
-                    new PipelineCacheActionRecord(level, uri, type, nameof(dedupManifestClient.PublishAsync), context));
-                PublishResult result = await clientTelemetry.MeasureActionAsync(
-                    record: uploadRecord,
-                    actionAsync: async () =>
+                // Tar the contents of the directory and create a tar, and then upload the tar.
+                /* string tempPath = Path.GetTempPath();
+                IEnumerable<string> absoluteFilesPaths = Directory.GetFiles(path, "*.*", SearchOption.AllDirectories);
```

Comment:
>GetFiles [](start = 67, length = 8)

There is a GetFileSystemEntries, but I'm not exactly sure how it works.

Also, I think if you pass a directory to TAR/7z it will process the whole contents of that directory, no?

Reply:
Removed the code. 
Also, I think if you pass a directory to TAR/7z it will process the whole contents of that directory, no? -> Yup, that's the plan. We are not supporting globing yet.